xdc-plugin/xdc.cc

```diff
@@ -236,21 +236,37 @@ struct SetProperty : public Pass {
 		if (args.size() < 1) {
 			log_error("set_property: Incorrect number of arguments.\n");
 		}
+
 		std::string parameter(args.at(0));
 		if (args.size() < 3 || args.at(2).size() == 0) {
 			log_error("set_property %s: Incorrect number of arguments.\n", parameter.c_str());
 		}
+
 		std::string port_name(args.at(2));
 		std::string value(args.at(1));
-		char port[128];
-		char bit[64];
-		if (sscanf(port_name.c_str(), "%[^[]%s", port, bit) == 2) {
-			port_name = std::string(port) + " " + std::string(bit);
+
+		auto port_signal = extract_signal(port_name);
+		std::string port(port_signal.first);
+		int port_bit = port_signal.second;
+
+		RTLIL::Wire* wire = design->top_module()->wire(RTLIL::escape_id(port));
+		if (wire == nullptr) {
+			log_error("Couldn't find port %s\n", port_name.c_str());
 		}
-		RTLIL::Module* top_module = design->top_module();
-		RTLIL::IdString port_id(RTLIL::escape_id(port_name));
+
+		if (!isInputPort(wire) && !isOutputPort(wire)) {
```

Comment:
Should handle inouts

Reply:
I will have to check, but I am pretty sure that for an inout both checks will return true, thus the condition will not be met and the error will not be raised.

Reply:
I checked and inout are handled fine. I added an inout to the test case.